pkg/pillar/zedUpload/datastore_aws.go

```diff
@@ -180,6 +182,7 @@ func (ep *AwsTransportMethod) processS3Download(req *DronaRequest) (error, int)
 	}
 
 	err := sc.DownloadFile(req.objloc, ep.bucket, req.name, req.sizelimit, prgChan)
+	req.cleanUpByteData()
```

Comment:
Why do you need this?

Reply:
since I was using the same req channel to send back the chunks to the caller, all the chunks have been sent as and when they on the writing on the same req, last req shouldn't have any byte data in it.
we need to clean it up before we confirming the full download.

Reply:
But why is it in ```processS3Download``` API, I think you are calling different API on the controller side?

Reply:
Sorry My bad! addressed please check.